EVSL_1.0/TESTS/GEN/KPM_MMRLanN.c

```diff
@@ -102,76 +102,73 @@ int main() {
     }
     fprintf(fstats, "MATRIX A: %s...\n", io.MatNam1);
     fprintf(fstats, "MATRIX B: %s...\n", io.MatNam2);
-    fprintf(fstats,
-            "Partition the interval of interest [%f,%f] into %d slices\n", a, b,
-            nslices);
+    fprintf(fstats, "Partition the interval of interest [%f,%f] into %d slices\n",
+            a, b, nslices);
     counts = malloc(nslices * sizeof(int));
     sli = malloc((nslices + 1) * sizeof(double));
     /*-------------------- Read matrix - case: COO/MatrixMarket formats */
     if (io.Fmt > HB) {
       ierr = read_coo_MM(io.Fname1, 1, 0, &Acoo);
       if (ierr == 0) {
         fprintf(fstats, "matrix read successfully\n");
-        nnz = Acoo.nnz;
         n = Acoo.nrows;
-        printf("size of A is %d\n", n);
-        printf("nnz of  A is %d\n", nnz);
       } else {
         fprintf(flog, "read_coo error for A = %d\n", ierr);
         exit(6);
       }
       ierr = read_coo_MM(io.Fname2, 1, 0, &Bcoo);
       if (ierr == 0) {
         fprintf(fstats, "matrix read successfully\n");
-        nnz = Bcoo.nnz;
-        n = Bcoo.nrows;
-        printf("size of B is %d\n", n);
-        printf("nnz of  B is %d\n", nnz);
+        if (Bcoo.nrows != n) {
+          return 1;
+        }
       } else {
         fprintf(flog, "read_coo error for B = %d\n", ierr);
         exit(6);
       }
       /*------------------ diagonal scaling for Acoo and Bcoo */
       sqrtdiag = (double *)calloc(n, sizeof(double));
-      extractDiag(&Bcoo, sqrtdiag);
-      diagScaling(&Acoo, &Bcoo, sqrtdiag);
-      /*-------------------- conversion from COO to CSR format */
+      /*------------------ conversion from COO to CSR format */
       ierr = cooMat_to_csrMat(0, &Acoo, &Acsr);
       ierr = cooMat_to_csrMat(0, &Bcoo, &Bcsr);
-    }
-    if (io.Fmt == HB) {
-      fprintf(flog, "HB FORMAT  not supported (yet) * \n");
+    } else if (io.Fmt == HB) {
+      fprintf(flog, "HB FORMAT not supported (yet) * \n");
       exit(7);
     }
-    alleigs = malloc(n * sizeof(double));
-    /*----------------  compute the range of the spectrum of B */
+
+    /*-------------------- diagonal scaling for L-S poly. approx. 
+     *                     of B^{-1} and B^{-1/2},
+     *                     which will be used in the DOS */
+    /*-------------------- sqrt of diag(B) */
+    extrDiagCsr(&Bcsr, sqrtdiag);
+    for (i=0; i<n; i++) {
+      sqrtdiag[i] = sqrt(sqrtdiag[i]);
+    }
+    /*-------------------- backup A and B */
+    csr_copy(&Acsr, &Acsr0, 1); /* 1 stands for memory alloc */
+    csr_copy(&Bcsr, &Bcsr0, 1);
+    /*-------------------- Scale A and B */
```

Comment:
save  the original A and B, and restore them after DOS: scaled matrices for DOS, original matrices for Lanczos solver